From: Ben Hutchings <ben@decadent.org.uk>
Date: Tue, 01 Dec 2015 02:21:58 +0000
Subject: af_unix: Avoid ABI changes

struct unix_sock is only allocated in af_unix so it's safe to add new
members at the end.

- Move recursion_level to the end of the structure.
- Hide recursion_level and peer_wake from genksyms.

---
--- a/include/net/af_unix.h
+++ b/include/net/af_unix.h
@@ -57,9 +57,11 @@ struct unix_sock {
         spinlock_t		lock;
 	unsigned int		gc_candidate : 1;
 	unsigned int		gc_maybe_cycle : 1;
-	unsigned char		recursion_level;
         wait_queue_head_t       peer_wait;
+#ifndef __GENKSYSMS__
+	unsigned char		recursion_level;
 	wait_queue_t		peer_wake;
+#endif
 };
 #define unix_sk(__sk) ((struct unix_sock *)__sk)
 
